src/neptune_pytorch/impl/__init__.py

```diff
@@ -152,11 +154,19 @@ def add_visualization_hook(self):
         def hook(module, input, output):
             if not self._is_viz_saved:
                 dot = torchviz.make_dot(output, params=dict(module.named_parameters()))
-                # Use tempfile correctly.
                 dot.format = "png"
-                dot.render(outfile="torch-viz.png")
-                self._namespace_handler["model"]["visualization"].upload("torch-viz.png")
-                self._is_viz_saved = True
+                # generate unique name so that multiple concurrent runs
+                # don't over-write each other.
+                viz_name = str(uuid.uuid4()) + ".png"
+                try:
+                    dot.render(outfile=viz_name)
+                    safe_upload_visualization(self._namespace_handler["model"], "visualization", viz_name)
+                except ExecutableNotFound:
+                    # This errors because `dot` renderer is not found even
+                    # if python binding of `graphviz` are available.
+                    warnings.warn("Skipping model visualization because no dot (graphviz) installation was found.")
+                finally:
+                    self._is_viz_saved = True
```

Comment:
Not sure about this part - `finally` will execute no matter what. If it's just about setting a variable, does it need to be in this block?

Reply:
Removed. Thanks!